DEX.md

```diff
@@ -0,0 +1,93 @@
+# What is a DEX?
+
+A Decentralized EXchange (DEX) is a smart contract that enables users to trade tokens without relying on a centralized intermediary. Instead, trades are executed directly on-chain through liquidity pools.
+Users willing to trade can sell their tokens of interest in return for another token.
+
+# Liquidity Pool
+
+Each DEX consists of one or more liquidity pools. A pool typically holds two tokens (say, $T_0$ and $T_1$). 
+Role of the users interacting with a liquidity pool can be:
+
+1. Liquidity Providers (LPs): Users who deposit tokens into pools are called liquidity providers.
+2. Traders: Users who swap one token for another. The exchange rate is determined by the ratio of tokens in the pool.
+3. Platform providers: Owners of the DEX.
+
+
+When a trade occurs, the trader pays a fee. This fee usually consists of:
+1. A platform fee that goes to the DEX protocol, and
+2. An LP fee that is distributed to liquidity providers, proportional to the amount of liquidity they contributed.
+
+Some platforms allow LPs to claim and reinvest their fees, effectively compounding their rewards.
+
+# Market Ratio
+
+A liquidity pool maintains a market ratio (MR) between its two tokens:
+
+$MR = \frac{reserve_0}{reserve_1}$
+
+Equivalently, it is the price of $T_1$ quoted in $T_0$. Intuitively, the higher the relative abundance of $T_0$ compared to $T_1$, the cheaper $T_1$ becomes in terms of $T_0$.
+
+When an LP deposits $amount_0$ and $amount_1$ to the pool (deposit ratio $DR = \frac{amount_0}{amount_1}$), pool takes from $amount_0$ and $amount_1$ according to the market ratio:
+
+1. if $DR == MR$ => pool takes $amount_0$ and $amount_1$
+2. if $DR >  MR$ => pool takes $amount_1 \times MR$ from $amount_0$ and $amount_0$
```

Comment:
what does the last part "and amount_0" mean? how much does it take from the other token?